File: trip-planner-server/src/main/java/com/tripplanner/module/day/service/DayService.java
package com.tripplanner.module.day.service;

import com.tripplanner.module.day.dto.DayCreateReq;
import com.tripplanner.module.day.dto.DayDetailResp;
import com.tripplanner.module.day.dto.DayResp;
import com.tripplanner.module.day.entity.TripDay;

import java.util.List;

public interface DayService {
    List<DayResp> listByTripId(Long userId, Long tripId);
    DayResp create(Long userId, Long tripId, DayCreateReq req);
    List<DayResp> generateDays(Long userId, Long tripId);
    DayDetailResp getDetail(Long userId, Long tripId, Long dayId);
    DayResp update(Long userId, Long tripId, Long dayId, DayCreateReq req);
    void delete(Long userId, Long tripId, Long dayId);
    TripDay findAndValidate(Long userId, Long dayId);
}

File: trip-planner-flutter/lib/services/day_service.dart
import 'package:flutter_riverpod/flutter_riverpod.dart';
import '../core/network/dio_client.dart';
import '../models/day.dart';

final dayServiceProvider = Provider<DayService>((ref) {
  return DayService(client: ref.read(dioClientProvider));
});

class DayService {
  final DioClient client;
  DayService({required this.client});

  Future<List<Day>> list(int tripId) async {
    final data = await client.get('/trips/$tripId/days');
    return (data as List<dynamic>).map((e) => Day.fromJson(e as Map<String, dynamic>)).toList();
  }

  Future<List<Day>> generate(int tripId) async {
    final data = await client.post('/trips/$tripId/days/generate');
    return (data as List<dynamic>).map((e) => Day.fromJson(e as Map<String, dynamic>)).toList();
  }

  Future<DayDetail> getDetail(int tripId, int dayId) async {
    final data = await client.get('/trips/$tripId/days/$dayId');
    return DayDetail.fromJson(data as Map<String, dynamic>);
  }

  Future<Day> create(int tripId, DayCreateReq req) async {
    final data = await client.post('/trips/$tripId/days', data: req.toJson());
    return Day.fromJson(data as Map<String, dynamic>);
  }

  Future<Day> update(int tripId, int dayId, DayCreateReq req) async {
    final data = await client.put('/trips/$tripId/days/$dayId', data: req.toJson());
    return Day.fromJson(data as Map<String, dynamic>);
  }

  Future<void> delete(int tripId, int dayId) async {
    await client.delete('/trips/$tripId/days/$dayId');
  }
}
